selfhost/typecheck.wyv

```diff
@@ -0,0 +1,70 @@
+module typecheck
+
+import types
+import bound
+import error
+import wyvern.collections.llist
+type List = llist.LinkedList
+
+type TPair //gamma is list of these
```

Comment:
I don't think you need a pair, just use the DeclType.  You can compute the unfold dynamically (and it doesn't exist for all DeclTypes, anyway, so it doesn't make sense to include it for all of them).